State by what component a system is not supported in a generic message

yarikoptic opened the pull request:
Otherwise it is hard to impossible (line numbers no longer match, pain) to determine who is complaining and why in a complex invocation composition

jwodder commented on src/datalad_installer.py:
```diff
@@ -2084,7 +2084,7 @@ def install_package(
     def assert_supported_system(self, **kwargs: Any) -> None:
         if not (ON_LINUX or ON_MACOS or ON_WINDOWS):
-            raise MethodNotSupportedError(f"{SYSTEM} OS not supported")
+            raise MethodNotSupportedError(f"{SYSTEM} OS not supported by {self}")
```
I think these should be `{self.NAME}` instead, as that shows the name used on the command line.

yarikoptic replied:
ok, let's do that

```suggestion
            raise MethodNotSupportedError(f"{SYSTEM} OS not supported by {self.NAME}")
```
?

while at it -- is there a way to include NAME of the component component (git-annex, etc) in addition?

jwodder replied:
You would have to pass `component` to `assert_supported_system()` on line 1468 and update all the `assert_supported_system()` definitions to include the string in the error message.

yarikoptic replied:
ok, let's then skip that for now  -- too much for `sed` ;)

yarikoptic committed:
[DATALAD RUNCMD] State by what a system is not supported in a generic message

Otherwise it is hard to impossible (line numbers no longer match, pain)
to determine who is complaining and why in a complex invocation composition

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "git-sedi '{{SYSTEM}} OS not supported' '{{SYSTEM}} OS not supported by {{self.NAME}}'",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^
